src/search/algorithms.clj

```diff
@@ -71,3 +71,43 @@
       (= current-state start-state) (reverse (conj path start-state))
       :else (recur (came-from current-state)
                    (conj path current-state)))))
+
+
+;;
+;; Another try at that a-star implementation
+;;
+(defn a-star [children-fn heuristic-fn start-state goal-state cost-fn]
+	(loop [frontier (pm/priority-map start-state 0)
+		cost-so-far {start-state 0}
+		came-from {}]
+
+		;; Check if we're done!
+		(if (or (empty? frontier)
+			(= (first (peek frontier)) goal-state))
+
```

Comment:
Why don't you check `max-states` anymore, or even pass it in? Given that there's the potential for infinite searching on some problems, that's a pretty useful, perhaps even crucial, feature. Seems odd that you'd remove it with no comment.